docs: add a starlark transformer tutorial (#2)
Akash-Nayak/hackfest: move2kube/exercise-3-customizations/exercise-3b-starlark-based-transformers/README.md

1. Let's start by going into the exercise directory. We will assume all commands are executed within this directory.
    ```console
    $ cd exercise-3-customizations/exercise-3b-starlark-based-transformers/
    ```

1. Lets use the [enterprise-app](https://github.com/konveyor/move2kube-demos/tree/main/samples/enterprise-app) as input.
    ```console
    $ ls ../../source/enterprise-app/
    README.md		config-utils		customers	docs			frontend		gateway			orders
    ```

1. Let's first run Move2Kube **without** any customizations. The output ingress does not have any annotations. We delete the output directory (`myproject/`) and files (`m2k-graph.json`, `m2kconfig.yaml`, `m2kqacache.yaml`) before moving on to the next step.
    ```console
    $ move2kube transform -s ../../source/enterprise-app/ --qa-skip && cat myproject/deploy/yamls/myproject-ingress.yaml && rm -rf myproject/ m2k*
    ```
    ```yaml
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      creationTimestamp: null
      labels:
        move2kube.konveyor.io/service: myproject
      name: myproject
    ```

> 1. We will use the starlark based custom transformer given [here](https://github.com/konveyor/move2kube-transformers/tree/main/add-custom-kubernetes-annotation). We will copy it into the `customizations` directory.

Reviewer: Aren't we using the custom transformer that we are providing in this exercise-3-b? We are not running the curl command to download the custom transformer from the move2kube-transformers repo and copying it to customizations folder.

Reviewer: ```suggestion
1. We will use the starlark based custom transformer given [here](https://github.com/konveyor/move2kube-transformers/tree/main/add-custom-kubernetes-annotation). We already have it in the `customizations` directory in this exercise.
```

    ```console
    $ ls customizations/add-custom-kubernetes-annotation/
    ```

1. Now lets transform with this customization. Specify the customization using the `--customizations` flag (short-hand `-c`). 
    ```console
    $ move2kube transform -s ../../source/enterprise-app/ -c customizations/ --qa-skip
    ```

Once the output is generated, we can observe from the below snippet of the ingress file (`myproject/deploy/yamls/myproject-ingress.yaml`) that there is an annotation for the ingress class added (`kubernetes.io/ingress.class: haproxy`):
```console
$ cat myproject/deploy/yamls/myproject-ingress.yaml
```
```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: haproxy
  creationTimestamp: null
  labels:
    move2kube.konveyor.io/service: myproject
  name: myproject
```

> ## Anatomy of ingress annotator transformer

Reviewer: Probably we can remove this Anatomy section as it's available in the full tutorial and we are also providing the custom transformer code along with the exercise. It will help in keeping the Readme file short.

This custom transformer uses a configuration yaml (`ingress-annotator.yaml`) and a starlark script (`ingress-annotator.star`) which adds the annotation to the ingress yaml. The contents of custom transformer are as shown below:
  ```console
  $ ls customizations/add-custom-kubernetes-annotation/
  ingress-annotator.star  ingress-annotator.yaml
  ```
The configuration yaml specifies that the custom transformer consumes and produces kubernetes yaml artifact type as shown in the `consumes` and `produces` section.
  ```console

>   $ cat customizations/add-custom-kubernetes-annotation/ingress-annotator.yaml

Reviewer: I think we changed the file name to transformer.yaml for all the custom transformers like https://github.com/konveyor/move2kube-transformers/tree/main/add-custom-kubernetes-annotation. So, do we want to rename the ingress-annotator.yaml to transformer.yaml.

        pathTemplateName = serviceName + yamlsBasePath
        pathTemplateName = pathTemplateName.replace("-", "")
        tplPathData = {'PathTemplateName': pathTemplateName}
        pathMappings.append({'type': 'PathTemplate', \
                            'sourcePath': "{{ OutputRel \"" + yamlsPath + "\" }}", \
                            'templateConfig': tplPathData})
        for f in fileList:
            filePath = fs.path_join(yamlsPath, f)
            s = fs.read(filePath)
            yamlData = yaml.loads(s)
            if yamlData['kind'] != 'Ingress':
                continue
            if 'annotations' not in yamlData['metadata']:
                yamlData['metadata']['annotations'] = {'kubernetes.io/ingress.class': 'haproxy'}
            else:
                yamlData['metadata']['annotations']['kubernetes.io/ingress.class'] = 'haproxy'
            s = yaml.dumps(yamlData)
            fs.write(filePath, s)
            pathMappings.append({'type': 'Default', \
                    'sourcePath': yamlsPath, \
                    'destinationPath': "{{ ." + pathTemplateName + " }}"})

    return {'pathMappings': pathMappings, 'artifacts': artifacts}
```

> The above steps can be replicated in the UI, by uploading the zip of the custom transformer as an input of type `customization`. You can get the zip of the source and customization by adding a `-z` to the end of the commands used in step 2 and step 4.

Reviewer: Can you please also add the zipped customization folder, so that can be directly used in the UI. We are already providing the zip of the source folder here https://github.com/konveyor/hackfest/blob/main/move2kube/source/enterprise-app.zip. Also, please update the Move2Kube UI related instruction accordingly.

Author: added the customizations.zip

Author: I have updated the last line about where to find the `customizations.zip` file
